main/src/ca/uwaterloo/flix/language/errors/RedundancyError.scala

```diff
@@ -338,6 +341,51 @@ object RedundancyError {
     })
   }
 
+  /**
+    * An error raised to indicate that an impure function expression is useless
+    * is statement position.
+    *
+    * @param tpe the type of the expression.
+    * @param loc the location of the expression.
+    */
+  case class UnderAppliedFunction(tpe: Type, loc: SourceLocation) extends RedundancyError {
+    def summary: String = "Under applied function. Missing function argument(s)?"
+
+    private def applicationAdvice(tpe: Type): String = {
+      val arguments = tpe.curriedArrowArgTypes
+      if (arguments.isEmpty) { // fallback message
+        "Missing function argument(s)?"
+      } else {
+        val argumentStrings = arguments.map(t => s"'${FormatType.formatWellKindedType(t)}'").mkString(", ")
+        val pluralS = if (arguments.sizeIs == 1) "" else "s"
+        s"Missing argument$pluralS of type: $argumentStrings."
```

Comment:
How about the format of this error message just becomes:

Missing argument(s) of type: Foo, Bar, Baz

?